src/toolkits/drawing_classifier/drawing_classifier.hpp

```diff
@@ -254,6 +261,12 @@ class EXPORT drawing_classifier : public ml_model_base {
     return variant_get_value<T>(get_state().at(key));
   }
 
+  /**
+   * for test svae & load purpose. well, nn_spec_ should not be exposed
+   * to user. I have to do this hack for test purpose
+   */
+  friend void init_dc_model_spec_for_test(drawing_classifier& dc);
+
```

Comment:
The pattern we're using in other toolkits is to provide a protected constructor that can directly initialize data members. Let's avoid the usage of `friend` if not necessary, since it's harder to reason about robustly

Reply:
But `friend` is only used for test purpose and it's very common in test case where a private member is involved. Thus, I think here it's okay.

I don't really think we should put anything that's not intended to be inherited in `protected` section.

Reply:
I should rephrase in this way: this function only makes sense for testing.

What this function does is to only initialize `nn_spec_`  in order to avoid segfault when calling save and load. 

But from the user's perspective, they should expect an exception to be thrown when `save` or `load` is called on an uninitialized model, which means the `init_train` is not called beforehand.

Reply:
rewrote the impl. Pls have a look.